ydb/core/tx/replication/service/transfer_writer.cpp

```diff
@@ -0,0 +1,708 @@
+    const TMessageOutputSpec& OutputSpec;
+
+public:
+    explicit TOutputListImpl(const TMessageOutputSpec& outputSpec, TWorkerHolder<IPullListWorker> worker)
+        : WorkerHolder_(std::move(worker))
+        , OutputSpec(outputSpec)
+        , Out(1)
+    {
+        Row.resize(1);
+    }
+
+public:
+    TOutputType* Fetch() override {
+        TBindTerminator bind(WorkerHolder_->GetGraph().GetTerminator());
+
+        with_lock(WorkerHolder_->GetScopedAlloc()) {
+            Out.Data.clear();
+
+            NYql::NUdf::TUnboxedValue value;
+
+            if (!WorkerHolder_->GetOutputIterator().Next(value)) {
+                return nullptr;
+            }
+
+            Out.Value = value.GetElement(0);
+            Out.Data.PushRow(&Out.Value, 1);
+
+            return &Out;
+        }
+    }
+
+private:
+    std::vector<NUdf::TUnboxedValue> Row;
+    TOutputType Out;
+};
+
+} // namespace
+
+} // namespace NKikimr::NReplication::NService
+
+
+template <>
+struct NYql::NPureCalc::TOutputSpecTraits<NKikimr::NReplication::NService::TMessageOutputSpec> {
+    static const constexpr bool IsPartial = false;
+
+    static const constexpr bool SupportPullListMode = true;
+
+    using TOutputItemType = NKikimr::NReplication::NService::TOutputType*;
+    using TPullStreamReturnType = THolder<IStream<TOutputItemType>>;
+    using TPullListReturnType = THolder<IStream<TOutputItemType>>;
+
+    static TPullListReturnType ConvertPullListWorkerToOutputType(
+        const NKikimr::NReplication::NService::TMessageOutputSpec& outputSpec,
+        TWorkerHolder<IPullListWorker> worker
+    ) {
+        return MakeHolder<NKikimr::NReplication::NService::TOutputListImpl>(outputSpec, std::move(worker));
+    }
+};
+
+
+namespace NKikimr::NReplication::NService {
+
+namespace {
+
+NYT::TNode CreateTypeNode(const TString& fieldType) {
+    return NYT::TNode::CreateList()
+        .Add("DataType")
+        .Add(fieldType);
+}
+
+NYT::TNode CreateOptionalTypeNode(const TString& fieldType) {
+    return NYT::TNode::CreateList()
+        .Add("OptionalType")
+        .Add(CreateTypeNode(fieldType));
+}
+
+
+void AddField(NYT::TNode& node, const TString& fieldName, const TString& fieldType) {
+    node.Add(
+        NYT::TNode::CreateList()
+            .Add(fieldName)
+            .Add(CreateOptionalTypeNode(fieldType))
+    );
+}
+
+NYT::TNode MakeOutputSchema(const TVector<TSchemaColumn>& columns) {
+    auto structMembers = NYT::TNode::CreateList();
+
+    for (const auto& column : columns) {
+        AddField(structMembers, column.Name, column.TypeName());
+    }
+
+    auto rootMembers = NYT::TNode::CreateList();
+    rootMembers.Add(
+        NYT::TNode::CreateList()
+            .Add(RESULT_COLUMN_NAME)
+            .Add(NYT::TNode::CreateList()
+                .Add("StructType")
+                .Add(std::move(structMembers)))
+    );
+
+    return NYT::TNode::CreateList().Add("StructType").Add(std::move(rootMembers));
```

Comment:
```suggestion
    return NYT::TNode::CreateList()
        .Add("StructType")
        .Add(std::move(rootMembers));
```
Выше так, можно и тут так же написать?